pkg/resource/pvc/scan.go

```diff
@@ -0,0 +1,96 @@
+package pvc
+
+import (
+	"errors"
+	"math"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apiresource "k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/kyma-project/kyma-metrics-collector/pkg/resource"
+)
+
+const (
+	// storageRoundingFactor rounds of storage to 32. E.g. 17 -> 32, 33 -> 64.
+	storageRoundingFactor = 32
+
+	// nfsPriceMultiplier is the factor by which the NFS PVCs are multiplied to compensate for the higher price.
+	nfsPriceMultiplier = 3
+
+	GiB = 1 << (10 * 3) //nolint:mnd // 1 GiB = 1024^3 bytes
+)
+
+var nfsLabels = map[string]string{
+	"app.kubernetes.io/component":  "cloud-manager",
+	"app.kubernetes.io/part-of":    "kyma",
+	"app.kubernetes.io/managed-by": "cloud-manager",
+}
+
+var _ resource.ScanConverter = &Scan{}
+
+type Scan struct {
+	pvcs corev1.PersistentVolumeClaimList
+}
+
+func (s *Scan) UM(duration time.Duration) (resource.UMMeasurement, error) {
+	return resource.UMMeasurement{}, nil
+}
+
+func (s *Scan) EDP() (resource.EDPMeasurement, error) {
+	edp := resource.EDPMeasurement{}
+
+	var errs []error
+
+	for _, pvc := range s.pvcs.Items {
+		// if the pvc has all labels defined in nfsLabels then it is an NFS PVC
+		if hasAllLabels(pvc.Labels, nfsLabels) {
+			if pvc.Status.Phase == corev1.ClaimBound {
+				currPVC := getSizeInGB(pvc.Status.Capacity.Storage())
+				// for NFS PVCs we multiply the used capacity by 3 to compensate for the higher price
+				nfsPVCStorage := currPVC * nfsPriceMultiplier
+				edp.ProvisionedVolumes.SizeGbTotal += nfsPVCStorage
+				edp.ProvisionedVolumes.SizeGbRounded += getVolumeRoundedToFactor(nfsPVCStorage)
+				edp.ProvisionedVolumes.Count += 1
+			}
+
+			continue
+		}
+
```

Comment:
Imho it would make sense to already extract the NFS logic into its own scanner/converter. We will extract it anyway into a scanner that uses the NFS resources as input.
This means we need to exclude NFS-PVCs here in this scanner and add another one just for the PVCs created by the NFS resources

Reply:
Let's introduce this change when we start scanning NFS CRs instead of the PVCs in order not to scrape them twice